sharder/deterministic.go

```diff
@@ -122,28 +123,50 @@ func (d *DeterministicSharder) Start() error {
 		if err != nil {
 			return errors.Wrap(err, "failed to get local interface list to initialize sharder")
 		}
+		localAddrCidrs := make([]string, len(localAddrs))
+		for i, addr := range localAddrs {
+			localAddrCidrs[i] = addr.String()
+		}
+
+		// If RedisIdentifier is an IP, add it to localAddrs.
+		redisIdentifier, err := d.Config.GetRedisIdentifier()
+		if err == nil {
```

Comment:
It's clearer, I think, if we don't even try if this wasn't specified.
```suggestion
		if err == nil && redisIdentifier != "" {
```

Reply:
I didn't know if I could assume `err == nil` means there exists a non-empty value.  If that's so then I will undo.

Reply:
Agreed.  I made this change.